src/components/Icon/Icon.tsx

```diff
@@ -44,6 +44,13 @@ export function Icon({source, color, backdrop, accessibilityLabel}: Props) {
     );
   }
 
+  if (color && !newDesignLanguage && isNewDesignLanguageColor(color)) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'You have selected a color meant to be used in the new design language but new design language is not enabled.',
+    );
+  }
+
```

Comment:
Is there any chance of this flooding console warnings in web?

Reply:
I don't think so. If the console is getting flooded with warnings then it would mean a ton of icons aren't rendering properly and that would be pretty alarming on its own